src/helpers/QueryParamRouter.js

```diff
@@ -0,0 +1,86 @@
+// @flow
+
+import * as React from 'react';
+import EventListener from 'react-event-listener';
+import { createMemoryHistory as createHistory } from 'history';
+import { Router } from 'react-router';
+
+type Props = {|
+  route: string,
+  children: React.Node,
+|};
+
+class QueryParamRouter extends React.Component<Props> {
+  history: {
+    push: string => void,
+  };
+  unlisten: () => void;
+
+  constructor(props: Props) {
+    super(props);
+
+    const history = createHistory({
+      initialEntries: [props.route],
+      initialIndex: 0,
+    });
+    this.unlisten = history.listen(this.onHistoryChange);
+    this.history = history;
+  }
+
+  componentDidMount() {
+    this.updateUrlParam(this.props.route);
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (prevProps.route !== this.props.route) {
+      this.history.push(this.props.route);
+      this.updateUrlParam(this.props.route);
+    }
+  }
+
+  componentWillUnmount() {
+    this.unlisten();
+  }
+
+  updateUrlParam = (pathname: string) => {
+    if (typeof window === 'undefined') {
+      return;
```

Comment:
When does this happen exactly? Is it because of tests?

Reply:
server side rendering :)

Reply:
hmm, gotcha. Maybe I'd add a comment here, or put it to a separate function.